src/main/java/org/opensearch/security/dlic/rest/api/AuthFailureListenersApiAction.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.dlic.rest.api;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.security.DefaultObjectMapper;
+import org.opensearch.security.dlic.rest.validation.EndpointValidator;
+import org.opensearch.security.dlic.rest.validation.RequestContentValidator;
+import org.opensearch.security.dlic.rest.validation.RequestContentValidator.DataType;
+import org.opensearch.security.securityconf.impl.CType;
+import org.opensearch.security.securityconf.impl.v7.ConfigV7;
+import org.opensearch.security.support.SecurityJsonNode;
+import org.opensearch.threadpool.ThreadPool;
+
+import static org.opensearch.rest.RestRequest.Method.DELETE;
+import static org.opensearch.rest.RestRequest.Method.GET;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+import static org.opensearch.security.dlic.rest.api.Responses.notFound;
+import static org.opensearch.security.dlic.rest.api.Responses.ok;
+import static org.opensearch.security.dlic.rest.api.Responses.response;
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+
+public class AuthFailureListenersApiAction extends AbstractApiAction {
+
+    public static final String IP_TYPE = "ip";
+
+    public static final String USERNAME_TYPE = "username";
+
+    public static final String NAME_JSON_PROPERTY = "name";
+
+    public static final String TYPE_JSON_PROPERTY = "type";
+    public static final String AUTHENTICATION_BACKEND_JSON_PROPERTY = "authentication_backend";
+    public static final String ALLOWED_TRIES_JSON_PROPERTY = "allowed_tries";
+    public static final String TIME_WINDOW_SECONDS_JSON_PROPERTY = "time_window_seconds";
+    public static final String BLOCK_EXPIRY_JSON_PROPERTY = "block_expiry_seconds";
+    public static final String MAX_BLOCKED_CLIENTS_JSON_PROPERTY = "max_blocked_clients";
+    public static final String MAX_TRACKED_CLIENTS_JSON_PROPERTY = "max_tracked_clients";
+
+    private static final List<Route> ROUTES = addRoutesPrefix(
+        ImmutableList.of(
+            new Route(GET, "/authfailurelisteners"),
```

Comment:
Do you plan to support GET with name as well?

Reply:
I figured it is not necessary, since the overall GET should return all instances anyway, and there is no additional information that can be returned if we build a different endpoint per auth failure listener, but I can do that if/when we feel we have enough functionality to justify that addition, wdyt?

Reply:
I agree. I think list is adequate for this API. In realistic scenarios, at most 2 auth_failure_listeners would be configured. Resolving this thread.